crypto/s2n_locking.c

```diff
@@ -75,7 +75,7 @@ S2N_RESULT s2n_locking_init(void)
 
     pthread_mutex_t *mutexes = S2N_MUTEXES(mutexes_mem);
     mutexes_count = 0;
-    for (size_t i = 0; i < num_locks; i++) {
+    for (size_t i = 0; i < (size_t) num_locks; i++) {
```

Comment:
Can you add a `RESULT_ENSURE_GTE(num_locks, 0)` right below the call to `CRYPTO_num_locks`?